### PR TITLE
feat: Sequence sender unit tests

### PR DESCRIPTION
**Pull request by rbpol**
Add unit tests for sequence sender

### CODE REVIEW

File: test/Makefile [hunk 1]
```diff
@@ -15,6 +15,9 @@ COMMON_MOCKERY_PARAMS=--disable-version-string --with-expecter --exported
 generate-mocks-sequencesender: ## Generates mocks for sequencesender, using mockery tool
 	rm -Rf ../sequencesender/txbuilder/mocks_txbuilder
 	export "GOROOT=$$(go env GOROOT)" && $$(go env GOPATH)/bin/mockery --all --case snake --dir ../sequencesender/txbuilder --output ../sequencesender/txbuilder/mocks_txbuilder --outpkg mocks_txbuilder ${COMMON_MOCKERY_PARAMS}
+	export "GOROOT=$$(go env GOROOT)" && $$(go env GOPATH)/bin/mockery --name=EthTxMngr --dir=../sequencesender --output=../sequencesender/mocks --outpkg=mocks --structname=EthTxMngrMock --filename=ethtxmanager.generated.go
```
Comment (rachit77) [inline test/Makefile:18]:
```suggestion
	export "GOROOT=$$(go env GOROOT)" && $$(go env GOPATH)/bin/mockery --name=EthTxMngr --dir=../sequencesender --output=../sequencesender/mocks --outpkg=mocks --structname=EthTxMngrMock --filename=mock_ethtxmanager.go
```

should we make file consistent with other mock files?

File: Makefile [hunk 2]
```diff
@@ -83,6 +90,13 @@ test-seq_sender:
 lint: ## Runs the linter
 	export "GOROOT=$$(go env GOROOT)" && $$(go env GOPATH)/bin/golangci-lint run --timeout 5m
 
+.PHONY: generate
+generate: generate-code-from-proto generate-mocks ## Generates mocks and other autogenerated types
+
+.PHONY: generate-mocks
+generate-mocks: ## Generates mocks
```
Comment (rachit77) [inline Makefile:97]: Do we still require this?
Since we have removed `.mockery.yml `file

Comment (rbpol) [reply Makefile]: Fixed

File: test/Makefile (hunk 1 above)
Comment (rbpol) [reply test/Makefile]: Done

File: sequencesender/ethtx_test.go [hunk 3]
```diff
@@ -0,0 +1,793 @@
...
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tt.expectEthTransactions, s.ethTransactions)
+			}
+		})
+	}
+}
+
+func Test_saveSentSequencesTransactions(t *testing.T) {
+	// TODO: Implement
```
Comment (goran-ethernal) [inline sequencesender/ethtx_test.go:792]: Are we going to address this through this PR?

File: sequencesender/sequencesender.go [hunk 4]
```diff
@@ -17,16 +19,42 @@ import (
 	"github.com/0xPolygonHermez/zkevm-ethtx-manager/ethtxmanager"
 	ethtxlog "github.com/0xPolygonHermez/zkevm-ethtx-manager/log"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
 )
 
 const ten = 10
 
+// EthTxMngr represents the eth tx manager interface
+type EthTxMngr interface {
```
Comment (goran-ethernal) [inline sequencesender/sequencesender.go:28]: Why not `EthTxManager`?

**Review by goran-ethernal — APPROVED** [review]

File: sequencesender/sequencesender.go (hunk 4 above)
Comment (rbpol) [reply sequencesender/sequencesender.go:29]: Just made it shorten :)

File: sequencesender/ethtx_test.go (hunk 3 above)
Comment (rbpol) [reply sequencesender/ethtx_test.go]: It is actually covered in another tests. Removed.

**Review by ToniRamirezM — APPROVED** [review]

File: sequencesender/sequencesender.go (hunk 4 above)
Comment (rbpol) [reply sequencesender/sequencesender.go]: Done

**Pull request merged**